hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/ReconContainerReportHandler.java

```diff
@@ -48,26 +46,16 @@ public ReconContainerReportHandler(NodeManager nodeManager,
   @Override
   public void onMessage(final ContainerReportFromDatanode reportFromDatanode,
                         final EventPublisher publisher) {
-
-    final ContainerReportsProto containerReport =
-        reportFromDatanode.getReport();
     ReconContainerManager containerManager =
         (ReconContainerManager) getContainerManager();
-
-    List<ContainerReplicaProto> reportsList = containerReport.getReportsList();
-    for (ContainerReplicaProto containerReplicaProto : reportsList) {
-      final ContainerID id = ContainerID.valueOf(
-          containerReplicaProto.getContainerID());
-      try {
-        containerManager.checkAndAddNewContainer(id,
-            containerReplicaProto.getState(),
-            reportFromDatanode.getDatanodeDetails());
-      } catch (IOException ioEx) {
-        LOG.error("Exception while checking and adding new container.", ioEx);
-      }
-      LOG.debug("Got container report for containerID {} ",
-          containerReplicaProto.getContainerID());
+    List<ContainerReplicaProto> containerReplicaProtoList =
+        reportFromDatanode.getReport().getReportsList();
+    try {
+      containerManager.checkAndAddNewContainerBatch(containerReplicaProtoList);
```

Comment:
Maybe we can set a batch size limit here to avoid the performance impact on SCM?